src/kvilib/net/KviSSL.cpp

```diff
@@ -863,11 +863,22 @@ void KviSSLCertificate::extractPubKeyInfo()
 
 void KviSSLCertificate::extractSerialNumber()
 {
+	m_szSerialNumber = KviCString();
 	ASN1_INTEGER * i = X509_get_serialNumber(m_pX509);
-	if(i)
-		m_iSerialNumber = ASN1_INTEGER_get(i);
-	else
-		m_iSerialNumber = -1;
+	if (i)
+	{
+		BIGNUM * bn = ASN1_INTEGER_to_BN(i, nullptr);
+		if (bn)
+		{
+			char *str = BN_bn2dec(bn);
```

Comment:
KVIrc coding style has a space between `*` and `str`.
Also, X509 serial numbers seem to be displayed in hex for humans, so I'd recommend using `BN_bn2hex` instead.